pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -262,13 +271,15 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, MultiRewa
         IERC20 pool,
         uint256 amount,
         uint256 deadline,
+        address account,
         address recipient,
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) external {
-        IERC20Permit(address(pool)).permit(msg.sender, address(this), amount, deadline, v, r, s);
-        stakeFor(pool, amount, recipient);
+    ) external nonReentrant {
+        require(account == recipient, "Cannot stake an accounts bpt to an alternate address");
```

Comment:
```suggestion
        // Force the  sender to the be the recipient to ensure the signature can only be used in a single way
        require(account == recipient, "Sender and recipient must equal for permit staking");
```

The original error message was not accurante and might be misleading, since it _is_ possible to stake to an alternate address - the restriction only shows up when permit staking.

Reply:
On a different note, we should also later review the fact that `amount` also has to be the same for both. This e.g. means that it is impossible to use `stakeWithPermit` while granting infinite allowance.

I imagine such a use case can be covered in some other contract by first doing a permit and then staking, but it still requires some analysis.